### PR TITLE
User Guide: Add documentation for subquery syntax

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/3091

 # Rationale for this change

We should tell users that we support subqueries and how to use them.

# What changes are included in this PR?

New section on subquery support.

# Are there any user-facing changes?

No

### CODE REVIEW

File: docs/source/user-guide/sql/subqueries.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Subqueries
+
+DataFusion supports `EXISTS`, `IN`, and Scalar Subqueries.
```
Comment (alamb) [inline docs/source/user-guide/sql/subqueries.md:22]:
```suggestion
DataFusion supports `EXISTS`, `NOT EXISTS`, `IN`, `NOT IN` and Scalar Subqueries.
```

File: docs/source/user-guide/sql/subqueries.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+```sql
+❯ select * from x y where not exists (select * from x where x.column_1 = y.column_1);
+0 rows in set.
+```
+
+## IN
+
+The `IN` syntax can be used to find all rows in a relation where a given expression's value can be found in the
+results of a correlated subquery. Only correlated subqueries are supported.
```
Comment (alamb) [inline docs/source/user-guide/sql/subqueries.md:63]:
```suggestion
results of a correlated subquery. 
```

I think non correlated IN / NOT IN is also supported:

```sql
❯ select * from foo;
+----+
| id |
+----+
| 1  |
| 3  |
| 2  |
| 10 |
| 8  |
+----+
5 rows in set. Query took 0.002 seconds.
❯ select * from foo where id in (select id from foo);
+----+
| id |
+----+
| 1  |
| 2  |
| 3  |
| 10 |
| 8  |
+----+
5 rows in set. Query took 0.007 seconds.
❯ select * from foo where id not in (select id from foo);
0 rows in set. Query took 0.007 seconds.
```

File: docs/source/user-guide/sql/subqueries.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
++----------+----------+
+| 1        | 2        |
++----------+----------+
+1 row in set.
+```
+
+## NOT IN
+
+The `NOT IN` syntax can be used to find all rows in a relation where a given expression's value can not be found in the
+results of a correlated subquery. Only correlated subqueries are supported.
```
Comment (alamb) [inline docs/source/user-guide/sql/subqueries.md:78]:
```suggestion
results of a correlated subquery.
```

**Review by alamb — APPROVED** [review]
Thanks @andygrove  -- looking very nice 👌

File: docs/source/user-guide/sql/subqueries.md (hunk 2 above)
Comment (andygrove) [reply docs/source/user-guide/sql/subqueries.md:63]: Ha. I am learning more about DataFusion! Documentation is good!

**Pull request merged**